### ISSUE TITLE
Security Issue

### ISSUE THREAD

**Issue by jhalt** [lucanaef/PGPro#18]
Hi @lucanaef,

I've come across what seems to be, in my estimation, a low-severity security issue.

Can you provide me with your pubkey and email for further dialog on this topic?

Thanks,
Jonathan

Edit: Alternatively, you can reach out to me directly via the contact info on my profile.

**Comment by lucanaef** [comment]
```
-----BEGIN PGP PUBLIC KEY BLOCK-----

xsFNBF2m03ABEADG6S02P1BH1aJOhqIEmHEbLi8G84Zspltix9bDT4HCPa8EXAxk
w+2YfIH5YPij5TfP32KOoxxEAKo9t+iPIJF83UHxoKTgAqBNtKn0cleqUhqB2E/U
SMnfVy7/QL6tXLpgRa7DRlqYIANOtBgmc21q/gSAKJLWMwIOV5ldgGnoxk9JI6hH
p2th+iQO6K+wpMgduxBDOkbBKHuEPYjkvavDFCgM3rykbex+vGSzBRsqhMQc91tf
9V79lmcLD1WUAYeuACS08oo1FNXMq4tXfTPeyEu0cIAyg2UiUi1vAO1FJnL5V4sI
k0+9UHsR/GBMg5sX32djYhSrS6Six5ZUuEbRP5VcafjgrGZXTMxoSbAFk0pRX/qo
wF0kH0YU2wZowp9dXvZZVxmzHW8zSHHSsP/5tt6W6LNQvVTGI9osRtz7YXpDyKi4
xfN+jVudZRWTolAh9fCz5F1oe71YcYgBDne2krnA0G0wtOS/+nPXwRH4AZkz0Pgn
rm/Rp9IRGmFrRM0asyss8hzExKsPRHru2LHfpj5DPM5FSgBcTdGhMwQM7o4IPYIh
OFThTWOBptRUkqUjcDxjk3noeUKu8xR+LNnWXveitxgOEyg424Ej85z3IQUIAZcW
ygBr4uk+76Ncf7KVufovFoZROcuQHzq1dcvwQdEomhtvniaSladJXlsYTwARAQAB
zRVQR1BybyA8ZGV2QHBncHJvLmFwcD7CwZQEEwEKAD4WIQQdNYzxCefBcfcIP/if
PzsPYvUStQUCXabTcAIbAwUJB4YfgAULCQgHAwUVCgkICwUWAgMBAAIeAQIXgAAK
CRCfPzsPYvUStafrD/wLJb+0sHoHPirtFAoGA1VjLnaX8kUnC+Wp5dpceqEZO1WP
eR3aSVK8lwb663BVww635idquySvsZcxWRfJz78NEOJUK4tH4klTuZMrOz5IB3o0
V+XjikbAyy01pT1KW9itGBEG8iJ/OqH6XSL0Hr5BFu+qe85c7liLm1Wc1U/0Y3pH
AoCMRe78NidOprS2GFAwD2fhpK6iy4A1YUOH0/Npi2Y/m62P5BgimeT7LkYBaQhX
pbprG4xKXixKrz9wDnMAwIEUhhApW2WkDV24+m/Ou0bgA9Q63ircViQGwoW+hKPh
JSdq04klpVBNKpEs+UiyILJFtem+qZu6/taCif7dItJNWwdmEMmE292pLJodDsYL
k1yhGBItFNdOFg8crTUVTsqF4wBNY0F5Pt3vnZCp1TZvA6WBysxwX7b9eF8RiEd0
q0r4WNne5WpqYC9ej/L3yGztBIX+JYt3Lj/Jb8fugbmrgorj/6N9UzvQFMiANFl9
7uBnLaMCDSWVfQa0srVrBHiOxEDSrvsf6FZ5CFYxMSw+UoYaqI2y+vlmnkqcNI9c
OY3Eig0NC0IGegnDYFrFWC8ALGg3LOawMdLQMeD1VZrjZJ5Fgip+yTekGRN33ZBu
PB01QRbzg6d2ATb9N3gQtcTxOLj6CJsXZCvku63gDM9QUsi+krYmyN30wWpzGs7B
TQRdptNwARAAvbh1v80f+81eIuyIqWa2q0MxjdptTWWeODVo0naecAFEPcSAeAjp
f5NgO5xiBWmXn6pbDzfRdl6SK3KlQowQmoSh1DGdbwxeLzhhjxH3bervrYL3Ddmj
0zrSHtkia4vCJXZJF0qnDM3libT2+bbcpfP5hCJVt/lYttik6TUrCIFRdUnrQhDc
ryMraS8IxsN9ybrUzrbpQsW2TEbGgFCROPuF5zBDmfmyXEqYfZyxKh11erKDSc8B
vRzBPM2Cl1QE6HrscmgkeuKjMgBE+f4nMWKGLDtcVBdnN4Yk4OMb0iqS/TQOk/ch
bCoHDiK1qaa5AN7s+K2KFHr5Umlb/dJ6vwdWkjPJ5imbv0bZ3Cwi+DjhSp48Hw49
8yd2H//Qk0VHySGrqOl/ifGxOlFnocHjEQgyL2PrmzohMmI3P0+lsqaAgemGzKPX
XaRX6ucuuX95IENsLd/KjbN+sCMyECVR8QRWsCyADqhx1z6rZMj6mwMfTb8J45Mv
FIYKcknY91xjRsT5jUZr7iX1uKaoieUEywZPS2yWbyq44jmb4LUF0Z0lEHpT4jU2
Pq/qmi7yudq0TaD/GMvF4wydbhQM2pxWDsPAvtnLfcNiKR43ZlgCGbLq4XiER8Ar
6EW9iClpnCYCRDBEflacWFCZZQIVvblxYVaVqfblJSimNIQMBipSTlEAEQEAAcLB
fAQYAQoAJhYhBB01jPEJ58Fx9wg/+J8/Ow9i9RK1BQJdptNwAhsMBQkHhh+AAAoJ
EJ8/Ow9i9RK1FjAQAI+cvmwn4mEr7sILMRbsNVtje67vj8J1comT42wKUiqUw5Ex
+FH+frOzh2To+7Y1Ui2pNjBUhXPsoXfo7Ni2RmfQ6u0Wur4pYErQI4jOLp9NC5Xt
1a0j4MkeYpqG7JmZ+/j8kt4RT/VLpnlVVaSA7xSakPnCSoMwwLU/+gEhU3f2zHX4
UsvLXddGSapktD/YMPnXWhYOoJAMwhOOgotVqhN3tsNwb5Zz2L/ELL0gIf9+Fo4S
RkMwhMUxf5XW3Mz5aC3mPGzDqve124+R2UlGWn2muQt9e7RNAEWuXvTnxbhuCsF1
SKOgPJ9fFeNopeDDpP8717MNEPLKzfnYZm5vQvAAmGLdOmcg4TUDBd5hLYCnjNpz
dKdTD1cp4V7mHkyBc123dUnyQb+LkLn71e6rbK4+JedKd5u+IdbS/JiOSssbE3Iz
pTlxjdMhUZzOqIfcU4RTNHAxUlV/dDFlhGyt2xC46oX6iQIGzm81DggYWUp9EmIp
U/zCQaNkkuO9YSrKvA3KbRuGLRZF+e5+yAHaRa4LJKDxkBhRvO1pFqlliojqQO5x
zvLsxRrImtFS/PaYDe+1EGBDt6jl0jZFe3+FugMEuHPDzJtelBN0lmpzNW36OIpG
CrnbByMEWOJ68IZ4R17G8dvtdMR16MIyOvOStbzSmZP8z0z063633TWTKePi
=2Rnf
-----END PGP PUBLIC KEY BLOCK-----
```


You can also find it here: [https://keys.openpgp.org/search?q=dev%40pgpro.app](https://keys.openpgp.org/search?q=dev%40pgpro.app) (verified keyserver)

**Comment by jhalt** [comment]
Thanks! I'll reach out to you shortly.

**Comment by jhalt** [comment]
This issue has been verifiably resolved in release 2020.4, which has been approved by Apple and is now publicly available.

I'll leave this issue open so it can be closed by the commit resolving this issue, or otherwise as the maintainers deem appropriate.

\- Jonathan

**Comment by lucanaef** [comment]
@jhalt reported a bug that caused - under certain conditions - exporting the private instead of the public key when using the share button on a contact detail view. After receiving the message, I've been able to reproduce, locate and fix that bug. The fixed version 2020.4 has been submitted to Apple a few days ago and should by now have propagated to most users. If anyone has further questions, please reach out! Many thanks again to @jhalt!
